### DISCUSSION TITLE
LiveCharts2 Axis Labels observablecollection

### DISCUSSION THREAD

**Discussion by g1deonr** [Live-Charts/LiveCharts2#1648] [Q&A]
I got my line chart to work dynamically using data in observablecollections, but the X axis Labels won't take an observablecollection.

Is there a way to do it already, or if not could this be a new feature that would make all source data types clear and consistent?

**Comment by beto-rodriguez** [comment]
Hello

Which platform is it? XAxes and YAxes properties are of type `IEnumerable<T>`, that should support observable collections.

Which is the error message you are getting?

**Comment by g1deonr** [comment]
Thanks for responding. It's C# WPF.

I'll post the error message details when I'm back at work on Monday.

**Comment by g1deonr** [comment]
Okay, Labels only accepts String, whereas my source data for the labels are Doubles. I don't think I can use MinLimit & MaxLimit which are Doubles because my axis is non-linear; increasing by 1% each point. Is there a neat way to use an ObservableCollection Double  as Labels?

Forgive me, I'm a noob (embedded programmer) who assumed that a chart axis could be numbers.

**Comment by beto-rodriguez** [comment]
> Okay, Labels only accepts String, whereas my source data for the labels are Doubles

You can try to convert them you string.

> I don't think I can use MinLimit & MaxLimit which are Doubles because my axis is non-linear

You can build custom scales:

https://livecharts.dev/docs/Maui/2.0.0-rc2/samples.axes.logarithmic

I will close this for now, since this is not an issue with the library, I'll move this to the Q&A section, feel free to reply there if you need further assistance.
